src/cpp/shared_core/system/User.cpp

```diff
@@ -52,54 +57,77 @@ struct User::Impl
 
       // Get the maximum size of a passwd for this system.
       long buffSize = ::sysconf(_SC_GETPW_R_SIZE_MAX);
-      if (buffSize < 0)
-         buffSize = 4096; // some systems return -1, be conservative!
 
-      std::vector<char> buffer(buffSize);
-      int result = in_getPasswdFunc(in_value, &pwd, &(buffer[0]), buffSize, &tempPtrPwd);
-      if (tempPtrPwd == nullptr)
+      if (buffSize < kGetPWMinBufSize)
```

Comment:
It seems reasonable that we could keep this init `buffSize` at the result returned from the `sysconf` call above as long as we don't get `-1`?

```suggestion
      if (buffSize < 0)
```

IMO `SC_GETPW_R_SIZE_MAX` is a super misleading name, I found this [documentation](https://pubs.opengroup.org/onlinepubs/9699919799.2008edition/functions/getpwuid.html):

> A call to sysconf(_SC_GETPW_R_SIZE_MAX) returns either -1 without changing errno or an initial value suggested for the size of this buffer

So it seems like if we get an actual value back, it should be a reasonable starting point, otherwise we can start with our hardcoded min.